internal/core/command/controller/messaging/router.go

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package messaging
+
+// MessagingRouter defines interface for Command Service to know
+// where to route the receiving device command response.
+type MessagingRouter interface {
+	// Get returns the responseTopicPrefix by RequestId
+	Get(string) string
+	// Set sets the responseTopicPrefix with RequestId as the key
+	Set(string, string)
+}
+
+func NewMessagingRouter() MessagingRouter {
```

Comment:
How is this Router determining which bus the response message goes to?

Reply:
My bad I forgot the most cirtical problem ...

A quick thought came to my mind is to maintain 2 maps internally :
```Go
func (r *MessagingRouter) Get(id string) (string, bool) {
    topic, ok := r.ExternalMap[id]
    if ok {
        return topic, true
    }

    topic, _ = r.InternalMap[id]
    return topic, false
}

func (r *MessagingRouter) Set(id string, topic string, source string) {
    switch source {
    case "internal":
        r.InternalMap[id] = topic
    case "external":
        r.ExternalMap[id] = topic
    }
}
```

Will try to improve it or find if there's better solution tomorrow ...

Reply:
Yes, that was what I had in mind.